nostr/relay.py

```diff
@@ -7,42 +8,52 @@
 from .message_type import RelayMessageType
 from .subscription import Subscription
 
+@dataclass
 class RelayPolicy:
-    def __init__(self, should_read: bool=True, should_write: bool=True) -> None:
-        self.should_read = should_read
-        self.should_write = should_write
-
+    should_read: bool = True
+    should_write: bool = True
+    
     def to_json_object(self) -> dict[str, bool]:
         return { 
             "read": self.should_read, 
             "write": self.should_write
         }
 
+
+
+@dataclass
+class RelayProxyConnectionConfig:
+    host: str
+    port: int
+    type: str
+
+
+
+@dataclass
 class Relay:
-    def __init__(
-            self, 
-            url: str, 
-            policy: RelayPolicy, 
-            message_pool: MessagePool,
-            subscriptions: dict[str, Subscription]={}) -> None:
-        self.url = url
-        self.policy = policy
-        self.message_pool = message_pool
-        self.subscriptions = subscriptions
-        self.lock = Lock()
-        self.ws = WebSocketApp(
-            url,
+    url: str
+    message_pool: MessagePool
+    policy: RelayPolicy = RelayPolicy()
+    ssl_options: dict = None
+    proxy_config: RelayProxyConnectionConfig = None
```

Comment:
```suggestion
    proxy_config: Union[None, RelayProxyConnectionConfig] = None
```

You need to type this properly. You can also use `Optional`

```suggestion
    proxy_config: Optional[RelayProxyConnectionConfig] = None
```

Reply:
@callebtc Curious, what does `Optional` accomplish? I haven't come across it before. I've viewed typing in python as almost more documentation than anything else. Seems like overkill when built-in types are fine being initialized to `None`.